client_proposal.go

```
@@ -45,15 +46,12 @@ func (c *Client) ProposeChannel(
 		Assets:   []channel.Asset{(*ethwallet.Address)(&c.cfg.AssetHolder.addr)},
 		Balances: [][]channel.Bal{initialBals.values},
 	}
-	prop := &client.ChannelProposal{
-		ChallengeDuration: uint64(challengeDuration),
-		Nonce:             nonce(),
-		ParticipantAddr:   c.wallet.NewAccount().Address(),
-		AppDef:            payment.AppDef(),
-		InitData:          &payment.NoData{},
-		InitBals:          alloc,
-		PeerAddrs:         []wallet.Address{c.onChain.Address(), (*ethwallet.Address)(&perunID.addr)},
-	}
+	prop, err := client.NewLedgerChannelProposal(
+		uint64(challengeDuration),
+		c.wallet.NewAccount().Address(),
+		alloc,
+		[]wire.Address{c.onChain.Address(), (*ethwallet.Address)(&perunID.addr)},
+		client.WithoutApp())
```

don't ignore the error here...

---

Oh, we dont have a linter here :laughing: I will run it locally